site/dev/common.sh

```diff
@@ -0,0 +1,228 @@
+create_latest () {
+  echo " --> create latest"
+
+  local ICEBERG_VERSION="$1"
+
+  # Ensure ICEBERG_VERSION is not empty
+  assert_not_empty "${ICEBERG_VERSION}"  
+
+  # Output the provided ICEBERG_VERSION for verification
+  echo "${ICEBERG_VERSION}"  
+
+  # Remove any existing 'latest' directory and recreate it
+  rm -rf docs/docs/latest/
+  mkdir docs/docs/latest/
+
+  # Create symbolic links and copy configuration files for the 'latest' documentation
+  ln -s "../${ICEBERG_VERSION}/docs" docs/docs/latest/docs
+  cp "docs/docs/${ICEBERG_VERSION}/mkdocs.yml" docs/docs/latest/
+
+  cd docs/docs/
+
+  # Update version information within the 'latest' documentation
+  update_version "latest"  
+  cd -
+}
+
+# Updates version information within the mkdocs.yml file for a specified ICEBERG_VERSION.
+# Arguments:
+#   $1: ICEBERG_VERSION - The version number used for updating the mkdocs.yml file.
+update_version () {
+  echo " --> update version"
+
+  local ICEBERG_VERSION="$1"
+
+  # Ensure ICEBERG_VERSION is not empty
+  assert_not_empty "${ICEBERG_VERSION}"  
+
+  # Update version information within the mkdocs.yml file using sed commands
+  if [ "$(uname)" == "Darwin" ]
+  then
+    sed -i '' -E "s/(^site\_name:[[:space:]]+docs\/).*$/\1${ICEBERG_VERSION}/" ${ICEBERG_VERSION}/mkdocs.yml
+    sed -i '' -E "s/(^[[:space:]]*-[[:space:]]+Javadoc:.*\/javadoc\/).*$/\1${ICEBERG_VERSION}/" ${ICEBERG_VERSION}/mkdocs.yml
+  elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]
+  then
+    sed -i'' -E "s/(^site_name:[[:space:]]+docs\/)[^[:space:]]+/\1${ICEBERG_VERSION}/" "${ICEBERG_VERSION}/mkdocs.yml"
+    sed -i'' -E "s/(^[[:space:]]*-[[:space:]]+Javadoc:.*\/javadoc\/).*$/\1${ICEBERG_VERSION}/" "${ICEBERG_VERSION}/mkdocs.yml"
+  fi
+
+}
+
+# Excludes versioned documentation from search indexing by modifying .md files.
+# Arguments:
+#   $1: ICEBERG_VERSION - The version number of the documentation to exclude from search indexing.
+search_exclude_versioned_docs () {
+  echo " --> search exclude version docs"
+  local ICEBERG_VERSION="$1"
+
+  # Ensure ICEBERG_VERSION is not empty
+  assert_not_empty "${ICEBERG_VERSION}"  
+
+  cd "${ICEBERG_VERSION}/docs/"
+
+  # Modify .md files to exclude versioned documentation from search indexing
+  python3 -c "import os
+for f in filter(lambda x: x.endswith('.md'), os.listdir()): lines = open(f).readlines(); open(f, 'w').writelines(lines[:2] + ['search:\n', '  exclude: true\n'] + lines[2:]);"
+
+  cd -
+}
+
+# Sets up local worktrees for the documentation and performs operations related to different versions.
+pull_versioned_docs () {
+  echo " --> pull versioned docs"
+  
+  # Ensure the remote repository for documentation exists and is up-to-date
+  create_or_update_docs_remote  
+
+  rm -r docs/docs
+
+  # Add local worktrees for documentation and javadoc from the remote repository
+  git worktree add docs/docs "${REMOTE}/docs"
+  git worktree add docs/javadoc "${REMOTE}/javadoc"
```

Comment:
I had still some old cruft on my worktree, so I needed to do this:
```suggestion
  git worktree add -f docs/docs "${REMOTE}/docs"
  git worktree add -f docs/javadoc "${REMOTE}/javadoc"
```